lua/entities/gmod_wire_expression2/core/custom/sv_selfaware2.lua

```diff
@@ -131,17 +174,19 @@ local GET_UDF_MODE = {
                 sig = string_sub(sig, idx + 1, -2) -- -2 in order to drop the ')' at the end.
                 local collection = res[name] or {}
                 res[name] = collection
-                -- MYTODO: entries [3] and [4] in the table (not sure yet how to obtain these)
+                -- TODO (Patch #2): entries [3] and [4] in the table (not sure yet how to obtain these)
                 collection[#collection + 1] = { [1] = sig, [2] = returnType or "" }
             end
-            return luaTableToE2(res, false) -- Convert to E2 table *without* array optimization.
```

Comment:
Oh riight, it was never enabled 😆
But even if you enabled it, that would have produced an incorrect table layout (it would try to create an array inside an array, and that is not going to be ok in the E2).
The localized `luaTableToE2` has fixed this by using a number instead of boolean, to represent a depth at which it will enable array optimization.
So, the depth/level of `0` means to enable array optimization on and past the second iteration :)